### PR TITLE
Ask confirmation when different user open reset password link

### PR DESCRIPTION
**Pull request by scruti**
When multiple users share a computer and one of them request the password reset, if when opening the link to reset the password another user is logged in the site, they will be ask to either:
- Continue as the logged user, cancelling the password reset.
- Proceed to reset the password as the user who requested it.

This behaviour imitates the current flow for completing the user registration when already signed in as another user.
Screenshot from 2020-03-30 12-22-57
Screenshot from 2020-03-30 12-23-16
Screenshot from 2020-03-30 12-23-34
Screenshot from 2020-03-30 12-23-45

### CODE REVIEW

**Comment by nasirkhanbeis** [comment]
Issue "already signed-in" flash message is not related this ticket its a regression and surfaces only when re-directing from /sign-in to home page when already sign-n. Ed raised a separate card for this.

File: psd-web/app/views/users/passwords/signed_in_as_another_user.html.erb [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+
+<div class="govuk-grid-row">
+  <div class="govuk-grid-column-two-thirds-from-desktop">
+    <h1 class="govuk-heading-l"><%= page_title %></h1>
+
+    <p class="govuk-body">Do you want to</p>
+
+    <ul class="govuk-list govuk-list--bullet">
+      <li><%= link_to "Continue as #{current_user.name}", root_path %></li>
+      <li><%= link_to "Reset password for #{@user_with_reset_token.name}",
```
Comment (edwardhorsford) [inline psd-web/app/views/users/passwords/signed_in_as_another_user.html.erb]: Both these links should have the `govuk-link` class added.

File: psd-web/app/controllers/users/passwords_controller.rb [hunk 2]
```diff
@@ -34,6 +35,11 @@ def edit
       end
     end
 
+    def sign_out_before_resetting_password
+      sign_out
```
Comment (frankieroberto) [inline psd-web/app/controllers/users/passwords_controller.rb]: @scruti I think we should avoid automatically signing users out from a `GET` request, as that introduces a minor security risk, whereby someone could force a user to sign out (eg by embedding an iframe or getting them to click a link) in order to more easily obtain their password when they sign back in (via some other mechanism).

Probably not a blocker, but would be good to fix if possible, eg by doing the sign out via a form doing a POST.

File: psd-web/app/views/users/passwords/signed_in_as_another_user.html.erb [hunk 3]
```diff
@@ -0,0 +1,18 @@
+<% page_title = "You are already signed in to the Product safety database" %>
+<% content_for :page_title, page_title %>
```
Comment (StupidCodeFactory) [inline psd-web/app/views/users/passwords/signed_in_as_another_user.html.erb]: Can you change this use the `page_title` helper?

File: psd-web/app/views/users/passwords/signed_in_as_another_user.html.erb [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
```
Comment (frankieroberto) [inline psd-web/app/views/users/passwords/signed_in_as_another_user.html.erb]: please use `<%= page_title(page_title) %>` helper here (mainly for consistency).

File: psd-web/app/views/users/passwords/signed_in_as_another_user.html.erb [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+
+<div class="govuk-grid-row">
+  <div class="govuk-grid-column-two-thirds-from-desktop">
+    <h1 class="govuk-heading-l"><%= page_title %></h1>
+
+    <p class="govuk-body">Do you want to</p>
```
Comment (frankieroberto) [inline psd-web/app/views/users/passwords/signed_in_as_another_user.html.erb]: Should this end with a colon?

```suggestion
    <p class="govuk-body">Do you want to:</p>
```

File: psd-web/app/controllers/users/passwords_controller.rb (hunk 2 above)
Comment (scruti) [reply psd-web/app/controllers/users/passwords_controller.rb]: Good point. Changed it.

File: psd-web/app/views/users/passwords/signed_in_as_another_user.html.erb [hunk 6]
```diff
@@ -9,9 +9,12 @@
...
     <ul class="govuk-list govuk-list--bullet">
       <li><%= link_to "Continue as #{current_user.name}", root_path, class: "govuk-link" %></li>
-      <li><%= link_to "Reset password for #{@user_with_reset_token.name}",
-              sign_out_before_resetting_password_path(reset_password_token: reset_password_token),
-              class: "govuk-link" %>
+      <li>
+        <%= form_with url: sign_out_before_resetting_password_path(reset_password_token: reset_password_token), method: :post do %>
+          <button class="button-link govuk-link">
+            <%= "Reset password for #{@user_with_reset_token.name}" %>
+          </button>
```
Comment (frankieroberto) [inline psd-web/app/views/users/passwords/signed_in_as_another_user.html.erb]: There's a helper for this:

```suggestion
            <%= govukButton(text:"Reset password for #{@user_with_reset_token.name}") %>
```

Comment (scruti) [reply psd-web/app/views/users/passwords/signed_in_as_another_user.html.erb]: @frankieroberto does this helper keep the button displayed as a link? Trying to keep the original list with 
- link a
- ink b

Comment (frankieroberto) [reply psd-web/app/views/users/passwords/signed_in_as_another_user.html.erb]: Oh no, I didn't spot that it was styled as a link. 🤦‍♂

**Pull request merged**